packages/lisk-p2p/src/peer.ts

```diff
@@ -278,48 +280,63 @@ export class Peer extends EventEmitter {
 	}
 
 	private _createOutboundSocket(): SCClientSocket {
-		return socketClusterClient.create({
+		const outboundSocket = socketClusterClient.create({
 			hostname: this._ipAddress,
 			port: this._wsPort,
 			query: JSON.stringify(this._nodeInfo),
 			autoConnect: false,
 		});
+
+		this._bindHandlersToOutboundSocket(outboundSocket);
+
+		return outboundSocket;
 	}
 
-	private _handlePeerInfo(request: ProtocolRPCRequest): void {
-		// Update peerInfo with the latest values from the remote peer.
-		// TODO ASAP: Validate and/or sanitize the request.data as a PeerInfo object.
-		/* tslint:disable:next-line: no-object-literal-type-assertion */
-		this._peerInfo = {
-			...this._peerInfo,
-			...request.data,
-		} as PeerInfo;
-		this.emit(EVENT_UPDATED_PEER_INFO, this._peerInfo);
+	// All event handlers for the outbound socket should be bound in this method.
+	private _bindHandlersToOutboundSocket(outboundSocket: SCClientSocket): void {
+		outboundSocket.on('connect', () => {
+			this.emit(EVENT_CONNECT_OUTBOUND);
+		});
 	}
 
-	private _startListeningForRPCs(
-		socket: SCServerSocket | SCClientSocket,
+	// All event handlers for the outbound socket should be unbound in this method.
+	/* tslint:disable-next-line:prefer-function-over-method*/
+	private _unbindHandlersFromOutboundSocket(
+		outboundSocket: SCClientSocket,
 	): void {
-		// If an existing listener is bound, remove it.
-		socket.off(REMOTE_EVENT_RPC_REQUEST, this._handleRPC);
-		socket.on(REMOTE_EVENT_RPC_REQUEST, this._handleRPC);
+		outboundSocket.off();
```

Comment:
should it be `outboundSocket.destroy()` ? Or, we can call `dropOutboundConnection()` from #1014 since we don't need this socket so its better to destroy?